### PR TITLE
Adds an oscal directory and GitHub Actions workflow for upstream OSCAL content

### PR DESCRIPTION
**Pull request by jpower432**
#### Description:

Adds initial vendor directory for OSCAL content with REV4 and REV5 NIST 800-53 catalogs and FedRAMP High profiles. Adds update-vendor.yaml for automated updates of content

#### Rationale:

Allow upstream OSCAL content to be updated, reviewed, and stored to aid to generating OSCAL component definitions.
Partially address issue #11106 

#### Review Hints:

This is part one to a two part feature add. The update-vendor.yaml was tested/verified locally using `act`.

### CODE REVIEW

**Comment by Mab879** [comment]
We store other vendor files (such as the STIG XML files) in `shared/references` that might be a better place for these files.

File: .github/workflows/update-vendor.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        working-directory: ./vendor
+      - name: Update content
+        uses: peter-evans/create-pull-request@v5.0.2
+        with:
+          base: master
+          branch: "vendor-update-${{ github.run_id }}"
+          delete-branch: true
+          commit-message: "Update vendored OSCAL content"
+          add-paths: |
+            vendor/
```
Comment (Mab879) [inline .github/workflows/update-vendor.yaml:55]: Please add a new line at the end of the file.

File: .github/workflows/update-vendor.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Update profiles
+        run: |
+          rm -rf "profiles/${{ matrix.variables.profile-name }}"
+          trestle import -f "${{ matrix.variables.profile-source }}" -o "${{ matrix.variables.profile-name }}"
+          trestle href --name "${{ matrix.variables.profile-name }}" -hr "trestle://catalogs/${{ matrix.variables.catalog-name }}/catalog.json"
+        working-directory: ./vendor
+      - name: Update content
+        uses: peter-evans/create-pull-request@v5.0.2
+        with:
+          base: master
```
Comment (Mab879) [inline .github/workflows/update-vendor.yaml:50]: Please add a useful title for these PRs.

https://github.com/peter-evans/create-pull-request#action-inputs

File: .github/workflows/update-vendor.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+name: Update vendored OSCAL content
```
Comment (Mab879) [inline .github/workflows/update-vendor.yaml:1]: Per the project [style gude](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#yaml) please use 4 space indention.

File: .github/workflows/update-vendor.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          rm -rf "profiles/${{ matrix.variables.profile-name }}"
+          trestle import -f "${{ matrix.variables.profile-source }}" -o "${{ matrix.variables.profile-name }}"
+          trestle href --name "${{ matrix.variables.profile-name }}" -hr "trestle://catalogs/${{ matrix.variables.catalog-name }}/catalog.json"
+        working-directory: ./vendor
+      - name: Update content
+        uses: peter-evans/create-pull-request@v5.0.2
+        with:
+          base: master
+          branch: "vendor-update-${{ github.run_id }}"
```
Comment (Mab879) [inline .github/workflows/update-vendor.yaml:51]: A body on the PR might be useful as well.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR.

I have a few minor points mainly around style.

**Comment by jpower432** [comment]
Thanks for the feedback @Mab879. I moved the directory to `shared/references/oscal`. Also changed some of the names to be more OSCAL specific since OSCAL is just one type of vendored file. Styling fixes also added in the last commit. Please let me know if any other changes are needed.

**Comment by jpower432** [comment]
Adding information on the last change. The workflow was originally pointing to the to the `oscal-content` v1.0.0 release, but there are newer version of the catalog. Added environment variables so the references are easier to change and updated to the latest commit on the repository before the update to OSCAL version 1.1.1. `compliance-trestle` does not support this version yet.

File: .github/workflows/update-oscal.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                  branch: "oscal-update-${{ github.run_id }}"
+                  delete-branch: true
+                  commit-message: "Update OSCAL content in shared/references/oscal"
+                  title: "Update upstream OSCAL content from usnistogv and GSA"
+                  body: |
+                      Updates upstream OSCAL content
+                      - usnistgov NIST 800-53 from "${{ matrix.variables.catalog-source }}"
+                      - GSA FedRAMP OSCAL profiles from "${{ matrix.variables.profile-source }}"
+                    
+                      Auto-generated by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action
```
Comment (Mab879) [inline .github/workflows/update-oscal.yml:63]: Could you also provide the path to this GitHub Actions file as well?

File: shared/references/oscal/REAMDE.adoc
Comment (Mab879) [inline shared/references/oscal/REAMDE.adoc:1]: Per the [style guide](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#asciidoc) please use Markdown unless there is a compelling reason.

**Pull request merged**